core/tests/dispatch_integration_tests.rs

```diff
@@ -1073,299 +1046,178 @@ fn network_status_port_established_lost_dropped_connection() {
     let mut net_cfg = NetworkConfig::default();
     net_cfg.set_max_connection_retry_attempts(6);
     net_cfg.set_connection_retry_interval(1000);
-    let local_system = system_from_network_config(net_cfg.clone());
-    let remote_system = system_from_network_config(net_cfg);
+    let pinger_system = system_from_network_config(net_cfg.clone());
+    let ponger_system = system_from_network_config(net_cfg);
 
     // Create a status_counter which will listen to the status port and count messages received
-    let (status_counter, _scf) = local_system.create_and_register(NetworkStatusCounter::new);
-    status_counter.on_definition(|c| {
-        local_system.connect_network_status_port(&mut c.network_status_port);
-    });
-    local_system.start(&status_counter);
+    let status_counter = start_status_counter(&pinger_system);
 
     // Create a pinger ponger pair such that the Network will be used.
-    let (ponger, _pof) = local_system.create_and_register(PongerAct::new_lazy);
-    let ponger_path = local_system
-        .register_by_alias(&ponger, "ponger")
-        .wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    local_system.start(&ponger);
-    let (pinger, _pif) =
-        remote_system.create_and_register(move || PingerAct::new_lazy(ponger_path));
-    let pinger_path = remote_system
-        .register_by_alias(&ponger, "ponger")
-        .wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    remote_system.start(&pinger);
-    // The systems establish a connection and the pings/pongs are sent
-    thread::sleep(Duration::from_millis(5000));
+    let (_, ponger_path) = start_ponger(&ponger_system, PongerAct::new_lazy());
+    let (_, pinger_done_future) =
+        start_pinger(&pinger_system, PingerAct::new_lazy(ponger_path.clone()));
 
-    // Inspect live sockets
-    // thread::sleep(Duration::from_millis(2 * 60 * 1000)); // "2MSL" to drop the socket
+    pinger_done_future
+        .wait_timeout(PINGPONG_TIMEOUT)
+        .expect("Pinger should complete");
 
     // Shutdown the remote system and wait for the connection to be lost and dropped by local_system
-    let _ = remote_system.kill_system();
-    thread::sleep(Duration::from_millis(2500));
+    let _ = ponger_system.kill_system();
+    thread::sleep(WAIT_TIME);
```

Comment:
I feel like this is still going to be flaky. Couldn't we say...try to rebind the second every 100ms and if it succeeds we know that the previous connection was dropped, or something? Instead of having a long unconditional waiting time.